424a425,506
> 
>   
> 
> 
>   @Test
>   public void testRenameDirectory() throws IOException {
>     Path src = new Path(TEST_ROOT_DIR, "dir1");
>     Path dst = new Path(TEST_ROOT_DIR, "dir2");
>     fileSys.delete(src, true);
>     fileSys.delete(dst, true);
>     assertTrue(fileSys.mkdirs(src));
>     assertTrue(fileSys.rename(src, dst));
>     assertTrue(fileSys.exists(dst));
>     assertFalse(fileSys.exists(src));
>   }
> 
>   
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
>   @Test
>   public void testRenameReplaceExistingEmptyDirectory() throws IOException {
>     Path src = new Path(TEST_ROOT_DIR, "dir1");
>     Path dst = new Path(TEST_ROOT_DIR, "dir2");
>     fileSys.delete(src, true);
>     fileSys.delete(dst, true);
>     assertTrue(fileSys.mkdirs(src));
>     writeFile(fileSys, new Path(src, "file1"), 1);
>     writeFile(fileSys, new Path(src, "file2"), 1);
>     assertTrue(fileSys.mkdirs(dst));
>     assertTrue(fileSys.rename(src, dst));
>     assertTrue(fileSys.exists(dst));
>     assertTrue(fileSys.exists(new Path(dst, "file1")));
>     assertTrue(fileSys.exists(new Path(dst, "file2")));
>     assertFalse(fileSys.exists(src));
>   }
> 
>   
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 
>   @Test
>   public void testRenameMoveToExistingNonEmptyDirectory() throws IOException {
>     Path src = new Path(TEST_ROOT_DIR, "dir1/dir2/dir3");
>     Path dst = new Path(TEST_ROOT_DIR, "dir1");
>     fileSys.delete(src, true);
>     fileSys.delete(dst, true);
>     assertTrue(fileSys.mkdirs(src));
>     writeFile(fileSys, new Path(src, "file1"), 1);
>     writeFile(fileSys, new Path(src, "file2"), 1);
>     assertTrue(fileSys.exists(dst));
>     assertTrue(fileSys.rename(src, dst));
>     assertTrue(fileSys.exists(dst));
>     assertTrue(fileSys.exists(new Path(dst, "dir3")));
>     assertTrue(fileSys.exists(new Path(dst, "dir3/file1")));
>     assertTrue(fileSys.exists(new Path(dst, "dir3/file2")));
>     assertFalse(fileSys.exists(src));
>   }
